protos/flyteidl/core/security.proto

```diff
@@ -69,6 +69,9 @@ message Identity {
     // oauth2_client references an oauth2 client. Backend plugins can use this information to impersonate the client when
     // making external calls.
     OAuth2Client oauth2_client = 3;
+
+    // exec_user_id references the id of the user who executes the task.
+    string exec_user_id = 4;
```

Comment:
@EngHabu what do you think about this?  We've had several requests over the past year and change to add some sort of human-level identity to the wf crd.  What do you think about this, just a free-form text?  presumably people can make this whatever they want, internal user id, email, jwt token possibly, encrypted encoded bytes, etc.

Reply:
ic... trying to reason about this... any planned use for it in flyte components?

Reply:
in our case, we pass the `exec_user_id` to the flyteplugins because our plugins require this info

Reply:
i think we should rename this field to just `identifier` or `user_identifier`.  @EngHabu ?

Reply:
`user_identifier` LGTM

Reply:
How does this work for subworkflows and launch plans triggered by workflows? Will the user identifier be passed down or will flytepropeller's app credentials be passed instead?

Reply:
Good point.  The audit trail should definitely encompass the fact that it came from propeller, but this field should continue to be the original user identifier string.  @ByronHsu this is possible with the design we're thinking right?  Or were you implying that this should be a list @katrogan ?

Reply:
I wasn't implying it should be a list but just wanted to understand the implications of hard-coding this to always reference a user if we anticipate app-driven identifiers

Reply:
i'm okay adding additional fields as they become necessary.

Reply:
Just to confirm, we will pass this along when Propeller launches LPs from a Workflow Execution...
What will the value be for scheduled workflows? some system/default value?